addons/website_sale/models/sale_order.py

```diff
@@ -170,7 +170,7 @@ def _get_pl_partner_order(self, country_code, show_visible, website_pl, current_
             pricelists |= all_pl.filtered(lambda pl: not show_visible or pl.selectable or pl.pricelist_id.id in (current_pl, order_pl)).mapped('pricelist_id')
 
         partner = self.env.user.partner_id
-        if not pricelists or partner.property_product_pricelist.id != website_pl:
+        if not pricelists or (partner_pl or partner.property_product_pricelist.id) != website_pl:
```

Comment:
Please document this change in the commit message.
This is business logic, and has nothing to do with the rest of the commit, which is purely technical!